internal/pkg/agent/application/actions/handlers/handler_action_upgrade.go

```diff
@@ -51,6 +51,25 @@ func (h *Upgrade) Handle(ctx context.Context, a fleetapi.Action, ack acker.Acker
 	if !runAsync {
 		return nil
 	}
+
+	if features.TamperProtection() {
```

Comment:
Why not the same pattern from the UNENROLL action with `tamperProtectionFn`?

I would prefer to see that they follow the same pattern when checking if tamper protection is enabled.